planning/mission_planner/src/lanelet2_plugins/default_planner.cpp

```diff
@@ -186,21 +194,114 @@ PlannerPlugin::MarkerArray DefaultPlanner::visualize(const HADMapRoute & route)
+  marker.points.push_back(
+    tier4_autoware_utils::createPoint(goal_footprint[3][0], goal_footprint[3][1], 0.0));
+  marker.points.push_back(marker.points.front());
+
+  msg.markers.push_back(marker);
+
+  return msg;
+}
+
+bool DefaultPlanner::check_goal_footprint(
+  const lanelet::ConstLanelet & current_lanelet,
+  const lanelet::ConstLanelet & combined_prev_lanelet,
+  const tier4_autoware_utils::Polygon2d & goal_footprint, double & next_lane_length,
+  const double search_margin)
+{
+  std::vector<tier4_autoware_utils::Point2d> points_intersection;
+
+  // check if goal footprint is in current lane
+  boost::geometry::intersection(
+    goal_footprint, combined_prev_lanelet.polygon2d().basicPolygon(), points_intersection);
+  if (points_intersection.empty()) {
+    return true;
+  }
+  points_intersection.clear();
+
+  // check if goal footprint is in between many lanelets
+  for (const auto & next_lane : routing_graph_ptr_->following(current_lanelet)) {
+    next_lane_length += lanelet::utils::getLaneletLength2d(next_lane);
+    lanelet::ConstLanelets lanelets;
+    lanelets.push_back(combined_prev_lanelet);
+    lanelets.push_back(next_lane);
+    lanelet::ConstLanelet combined_lanelets = combine_lanelets(lanelets);
+
+    // if next lanelet length longer than vehicle longitudinal offset
+    if (vehicle_info_.max_longitudinal_offset_m + search_margin < next_lane_length) {
+      next_lane_length -= lanelet::utils::getLaneletLength2d(next_lane);
+      boost::geometry::intersection(
+        goal_footprint, combined_lanelets.polygon2d().basicPolygon(), points_intersection);
+      if (points_intersection.empty()) {
+        return true;
+      }
+      points_intersection.clear();
+    } else {  // if next lanelet length shorter than vehicle longitudinal offset -> recursive call
+      if (!check_goal_footprint(next_lane, combined_lanelets, goal_footprint, next_lane_length)) {
+        continue;
+      } else {
+        return true;
+      }
+    }
+    next_lane_length = 0.0;
+  }
+  return false;
+}
+
+bool DefaultPlanner::is_goal_valid(
+  const geometry_msgs::msg::Pose & goal, lanelet::ConstLanelets path_lanelets)
 {
+  const auto logger = node_->get_logger();
   lanelet::Lanelet closest_lanelet;
   if (!lanelet::utils::query::getClosestLanelet(road_lanelets_, goal, &closest_lanelet)) {
     return false;
   }
+
+  const auto local_vehicle_footprint = create_vehicle_footprint(vehicle_info_);
+  tier4_autoware_utils::LinearRing2d goal_footprint =
+    transformVector(local_vehicle_footprint, tier4_autoware_utils::pose2transform(goal));
+  pub_goal_footprint_marker_->publish(visualize_debug_footprint(goal_footprint));
+  const auto polygon_footprint = convert_linear_ring_to_polygon(goal_footprint);
+
+  double next_lane_length = 0.0;
+  // combine calculated route lanelets
+  lanelet::ConstLanelet combined_prev_lanelet = combine_lanelets(path_lanelets);
+
+  // check if goal footprint exceeds lane and isn't in parking_lot
```

Comment:
> and isn't in parking_lot

I feel `when the goal is in parking_lot` makes us easier to understand the intention of this if condition. What do you think.